### PR TITLE
tool: elektrad - add documentation endpoint

### PR DESCRIPTION
**Pull request by raphi011**
## Basics

These points need to be fulfilled for every PR:

- [ ] Short descriptions of your changes are in the release notes
      (added as entry in `doc/news/_preparation_next_release.md` which
      contains `_(my name)_`)
      **Please always add something to the release notes.**
- [ ] Details of what you changed are in commit messages
      (first line should have `module: short statement` syntax)
- [ ] References to issues, e.g. `close #X`, are in the commit messages.
- [ ] The buildservers are happy.
- [ ] The PR is rebased with current master.

If you have any troubles fulfilling these criteria, please write
about the trouble as comment in the PR. We will help you.
But we cannot accept PRs that do not fulfill the basics.

## Checklist

Check relevant points but **please do not remove entries**.
For docu fixes, spell checking, and similar none of these points below
need to be checked.

- [ ] I added unit tests for my code
- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I added code comments, logging, and assertions as appropriate (see [Coding Guidelines](https://master.libelektra.org/doc/CODING.md))
- [ ] I updated all meta data (e.g. README.md of plugins and [METADATA.ini](https://master.libelektra.org/doc/METADATA.ini))
- [ ] I mentioned every code not directly written by me in [THIRD-PARTY-LICENSES](doc/THIRD-PARTY-LICENSES)

## Review

Reviewers will usually check the following:

- [ ] Documentation is introductory, concise, good to read and describes everything what the PR does
- [ ] Examples are well chosen and understandable
- [ ] Code is conforming to [our Coding Guidelines](https://master.libelektra.org/doc/CODING.md)
- [ ] APIs are conforming to [our Design Guidelines](https://master.libelektra.org/doc/DESIGN.md)
- [ ] Code is consistent to [our Design Decisions](https://master.libelektra.org/doc/decisions)

## Labels

If you are already Elektra developer:

- Add the "work in progress" label if you do not want the PR to be reviewed yet.
- Add the "ready to merge" label **if the basics are fulfilled** and you also
  say that everything is ready to be merged.

### CODE REVIEW

**Comment by raphi011** [comment]
> Do we update the site https://elektrad.docs.apiary.io/?

Not at the moment, do you prefer updating this site or rendering / redirecting to the markdown instead

**Comment by markus2330** [comment]
I would prefer returning the text, even if it is not rendered. Then we should also get rid of all references to http://docs.elektrad.apiary.io in our docu, though. Which dependencies does elektrad have at the moment? If we do not have any, we should not get one just for the markdown renderer.

**Comment by mpranj** [comment]
Is this ready or still WIP?

**Comment by raphi011** [comment]
> Is this ready or still WIP?

since this MR was approved I think it's ready to merge. Replacing the redirect with a textual documentation can be done in a separate PR if still needed

**Comment by mpranj** [comment]
Sorry, I overlooked it because the `ready to merge` label was missing.

Sometimes a PR gets reviewed multiple times and it's not always clear whether the author will still push something.

Thanks for resolving the conflicts! I will merge as soon as the build passes.

File: src/tools/elektrad/doc_handler.go [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+)
+
+// getDocHandler redirects to the API documentation.
+//
+// Response Code:
+//		301 Moved Permanently Redirect to the documentation.
+//
+// Example: `curl localhost:33333`
+func (s *server) getDocHandler(w http.ResponseWriter, r *http.Request) {
+	http.Redirect(w, r, "https://elektrad.docs.apiary.io", 301)
```
Comment (markus2330) [inline src/tools/elektrad/doc_handler.go:14]: Isn't the info there outdated?

Comment (raphi011) [reply src/tools/elektrad/doc_handler.go:14]: well the functionality stayed the samed so i think it should still apply?

Comment (markus2330) [reply src/tools/elektrad/doc_handler.go:14]: You know best if it still applies : wink: What about linking to https://master.libelektra.org/doc/api_blueprints/elektrad.apib instead?

**Comment by mpranj** [comment]
Thank you!

**Pull request merged**